recipes/c-client/all/conanfile.py

```diff
@@ -0,0 +1,135 @@
+import functools
+import os
+import stat
+
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+
+class CclientConan(ConanFile):
+    name = "c-client"
+    description = "University of Washington IMAP toolkit"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/uw-imap/imap"
+    topics = "imap"
+    license = "Apache-2.0"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        # FIXME: can this library be made shared?
```

Comment:
but you did it, with .def file, right?

Reply:
Actually, no. I just included the def file from PHP before adding this fixme and actually compiling it. The library can't actually be made shared, because it needs symbols defined in the consuming binary. See the test source. Unless I'm misunderstanding something here. cc @cmb69 

I guess I can remove this and the def file.

Reply:
Or well, I guess just `extern`ing them should be enough to make it work as a shared library, but is that really useful here?

Reply:
> I just included the def file from PHP […]

Hmm, which .def file?

Regarding shared builds: these are not necessary for PHP on Windows, but may make sense for other use cases.

Reply:
Okay, sorry. I mixed up this recipe with qdbm's, where I did in fact take the def from winlibs. The def file for this recipe was generated by invoking CMake's internal command to scan and export all symbols.

Reply:
I would say, either remove the `.def` file from recipe, or add an actual (working) shared option.
in general, if an upstream library doesn't support some configuration, it's fine if conan recipe doesn't support it as well.
especially, if you don't need this configuration, it's totally acceptable.
someone who actually need shared builds, may contribute them later, for the first version of the recipe, there is no strict requirement to support everything.
especially, given how long is `.def` file.

Reply:
Yeah, shared build doesn't work without changing the library to use `dl` style runtime loading of the symbols, so shared build is totally out the window it seems. I will remove things related to that.

Reply:
okay, please remove `.def` files , FIXME comments and any other leftover related to the (unfinished) efforts to make shared builds work